app/code/core/Mage/Adminhtml/Block/System/Currency/Rate/Matrix.php

```diff
@@ -71,7 +71,7 @@ protected function _prepareRates($array)
 
         foreach ($array as $key => $rate) {
             foreach ($rate as $code => $value) {
-                $parts = explode('.', $value);
+                $parts = explode('.', (string)$value);
```

Comment:
We expect `$value` to be a number, if it's `null`, set it to 0.
```suggestion
                $value ?? $value = 0;
                $parts = explode('.', $value);
```
The null coalescing operator above may not be a cup of tea for most, but I like it. See https://onlinephp.io/c/9a9a93. A conventional way:
```suggestion
                if ($value === null) {
                     $value = 0;
                }
                $parts = explode('.', $value);
```

Reply:
wouldn't it be better `$value ?? ''`? why exploding integer zero?

Reply:
$value is expected to be a number.

Reply:
explode expects a string https://www.php.net/explode

Reply:
`$value === null ? '' : explode(....`